client/lib/feed-store/index.js

```diff
@@ -89,6 +94,17 @@ function receiveFeeds( newFeeds ) {
 	} ) );
 }
 
+function discoverFeeds( data, feedUrl ) {
+	if ( typeof data !== 'undefined' ) {
+		data.feeds.map( function( feed ) {
```

Comment:
need to check that data and data.feeds is valid before calling map on it. Might use lodash/map instead of the native map too do easily handle nulls.